script/testing/util/test_server_v2.py

```diff
@@ -0,0 +1,223 @@
+                self.print_output(self.db_output_file)
+                if attempt + 1 == constants.DB_START_ATTEMPTS:
+                    raise
+                traceback.print_exc(file=sys.stdout)
+                pass
+        # FOR
+        return
+
+    def wait_for_db(self):
+        """ Wait for the db server to come up """
+
+        # Check that PID is running
+        if not check_pid(self.db_process.pid):
+            raise RuntimeError("Unable to find DBMS PID {}".format(
+                self.db_process.pid))
+
+        # Wait a bit before checking if we can connect to give the system time to setup
+        time.sleep(constants.DB_START_WAIT)
+
+        # flag to check if the db is running
+        is_db_running = False
+
+        # Keep trying to connect to the DBMS until we run out of attempts or we succeeed
+        for i in range(constants.DB_CONNECT_ATTEMPTS):
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            try:
+                s.connect((self.db_host, int(self.db_port)))
+                s.close()
+                print("Connected to server in {} seconds [PID={}]".format(
+                    i * constants.DB_CONNECT_SLEEP, self.db_process.pid))
+                is_db_running = True
+                break
+            except:
+                if i > 0 and i % 20 == 0:
+                    print("Failed to connect to DB server [Attempt #{}/{}]".
+                          format(i, constants.DB_CONNECT_ATTEMPTS))
+                    # os.system('ps aux | grep terrier | grep {}'.format(self.db_process.pid))
+                    # os.system('lsof -i :15721')
+                    traceback.print_exc(file=sys.stdout)
+                time.sleep(constants.DB_CONNECT_SLEEP)
+                continue
+
+        if not is_db_running:
+            msg = "Unable to connect to DBMS [PID={} / {}]"
+            status = "RUNNING"
+            if not check_pid(self.db_process.pid):
+                status = "NOT RUNNING"
+            msg = msg.format(self.db_process.pid, status)
+            raise RuntimeError(msg)
+        return
+
+    def stop_db(self):
+        """ Stop the Db server and print it's log file """
+        # get exit code, if any
+        self.db_process.poll()
+        if self.db_process.returncode is not None:
+            # Db terminated already
+            self.db_output_fd.close()
+            self.print_output(self.db_output_file)
+            msg = "DB terminated with return code {}".format(
+                self.db_process.returncode)
+            raise RuntimeError(msg)
+
+        # still (correctly) running, terminate it
+        self.db_process.terminate()
+        return
+
+    def print_output(self, filename):
+        """ Print out contents of a file """
+        fd = open(filename)
+        lines = fd.readlines()
+        for line in lines:
+            print(line.strip())
+        fd.close()
+        return
+
+    def run_test(self, test_case: TestCase):
```

Comment:
wait i prefer to put it here. since the `TestServer` class should be the most generic one that could fit for all tests, and the `TestCase` may not have been introduced to the junit tests yet.